tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -575,3 +587,76 @@ ROOT::Experimental::Detail::RPageSourceFile::LoadCluster(DescriptorId_t clusterI
       cluster->SetColumnAvailable(colId);
    return cluster;
 }
+
+
+void ROOT::Experimental::Detail::RPageSourceFile::UnzipClusterImpl(RCluster *cluster)
+{
+   RNTupleAtomicTimer timer(fCounters->fTimeWallUnzip, fCounters->fTimeCpuUnzip);
+   fTaskScheduler->Reset();
+
+   const auto clusterId = cluster->GetId();
+   const auto &clusterDescriptor = fDescriptor.GetClusterDescriptor(clusterId);
+
+   std::vector<std::unique_ptr<RColumnElementBase>> allElements;
+
+   const auto &columnsInCluster = cluster->GetAvailColumns();
+   for (const auto columnId : columnsInCluster) {
+      const auto &columnDesc = fDescriptor.GetColumnDescriptor(columnId);
+
+      allElements.emplace_back(RColumnElementBase::Generate(columnDesc.GetModel().GetType()));
+
+      const auto &pageRange = clusterDescriptor.GetPageRange(columnId);
+      std::uint64_t pageNo = 0;
+      std::uint64_t firstInPage = 0;
+      for (const auto &pi : pageRange.fPageInfos) {
+         ROnDiskPage::Key key(columnId, pageNo);
+         auto onDiskPage = cluster->GetOnDiskPage(key);
+         R__ASSERT(onDiskPage);
+         R__ASSERT(onDiskPage->GetSize() == pi.fLocator.fBytesOnStorage);
+
+         auto taskFunc =
```

Comment:
This seems to be an almost copy of `RPageSourceFile::PopulatePageFromCluster`. Can you refactor to reuse the same code in both cases?

Reply:
Addressed in #7025